infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -0,0 +1,376 @@
+
+    @Configurable(description = "Azure virtual machine size type (by default: 'Standard_D1_v2'")
+    protected String vmSizeType = null;
+
+    @Configurable(description = "The virtual machine Username")
+    protected String vmUsername = null;
+
+    @Configurable(description = "The virtual machine Password")
+    protected String vmPassword = null;
+
+    @Configurable(description = "A public key to allow SSH connection to the VM")
+    protected String vmPublicKey = null;
+
+    @Configurable(description = "The Azure resourceGroup to use (if not specified, the one from the image will be used")
+    protected String resourceGroup = null;
+
+    @Configurable(description = "The Azure Region to use (if not specified, the one from the image will be used")
+    protected String region = null;
+
+    @Configurable(description = "Total instance to create")
+    protected int numberOfInstances = 1;
+
+    @Configurable(description = "Total nodes to create per instance")
+    protected int numberOfNodesPerInstance = 1;
+
+    @Configurable(description = "Command used to download the worker jar")
+    protected String downloadCommand = generateDefaultDownloadCommand();
+
+    @Configurable(description = "Optional network CIDR to attach with new VM(s) (by default: '10.0.0.0/24')")
+    protected String privateNetworkCIDR = null;
+
+    @Configurable(description = "Optional flag to specify if the public IP(s) of the new VM(s) must be static ('true' by default)")
+    protected boolean staticPublicIP = true;
+
+    @Configurable(description = "Additional Java command properties (e.g. \"-Dpropertyname=propertyvalue\")")
+    protected String additionalProperties = "-Dproactive.useIPaddress=true";
+
+    protected ConnectorIaasController connectorIaasController = null;
+
+    protected final Map<String, Set<String>> nodesPerInstances;
+
+    /**
+     * Default constructor
+     */
+    public AzureInfrastructure() {
+        nodesPerInstances = Maps.newConcurrentMap();
+    }
+
+    @Override
+    public void configure(Object... parameters) {
+
+        logger.info("Validating parameters : " + parameters);
+        validate(parameters);
+
+        this.clientId = parameters[0].toString().trim();
+        this.secret = parameters[1].toString().trim();
+        this.domain = parameters[2].toString().trim();
+        this.subscriptionId = parameters[3].toString().trim();
+        this.authenticationEndpoint = parameters[4].toString().trim();
+        this.managementEndpoint = parameters[5].toString().trim();
+        this.resourceManagerEndpoint = parameters[6].toString().trim();
+        this.graphEndpoint = parameters[7].toString().trim();
+        this.rmHostname = parameters[8].toString().trim();
+        this.connectorIaasURL = parameters[9].toString().trim();
+        this.image = parameters[10].toString().trim();
+        this.vmSizeType = parameters[11].toString().trim();
+        this.vmUsername = parameters[12].toString().trim();
```

Comment:
indexes created in: 06b250351a3cac447f5c0b0272c314c5bd90a0cb
Regarding the map, this requires deep modification inside the whole project, issue created: https://github.com/ow2-proactive/nodesource-addons/issues/32